### DISCUSSION TITLE
Capture select_or_add_multiple output dynamically

### DISCUSSION THREAD

**Discussion by courtiol** [rubenarslan/formr.org#586] [Q&A] [answered]
To capture what respondents fill within an item of type `text` on name *input*, the following works fine:
```js
<script>

window.onload = function () {

    const reader = document.querySelector('[name="input"]');
    const writer = document.querySelector('#result');

    reader.addEventListener("change", (event) => {
        textread = event.target.value;
        writer.textContent = textread;
    });
};

</script>
```
I use it to define text in an item of type note. 
```js
## text is <span id="result"></span>
```

My issue is that I am trying to adjust that particular piece of code for capturing the content of an item of type `select_or_add_multiple` and I don't seem to manage...

I am not familiar at all with javascript and I did not figure out how to detect the name and nature of objects created by website. Here I suspect that `select_or_add_multiple` should be handled as an array but I am not even sure. 
I tried debugging features with no avail (I cannot get `console.log()` to work when it is declared within the listener).

Any help would be deeply appreciated.

Here is a simple example of survey sheet I am using to experiment:
https://docs.google.com/spreadsheets/d/12DZM1Sinen3ZyTgLeWF1riZCDjz0q-2kGigaytQdMwk/edit?usp=sharing

**Comment by rubenarslan** [comment, marked as answer]
Here's a working version as far as I understood what you need.
https://docs.google.com/spreadsheets/d/1jKrzbxpCTq6UD2D3FtR4tHhNMTbE-G-IS3qZOQRpqN0/edit?gid=0#gid=0
The dropdown select or add multiple is backed by select2 and the value is stored in a hidden input type text. If you use jQuery, which is enabled in formr, it is easier; there's some synergy there.

```javascript
$(document).ready(function () {
    $('input[name="input"]').on('input', function () {
        var textread = $(this).val();
        var match = textread.match(/\s\d+/);
        var numberread = match ? match[0] : '';
        $('#result').text(numberread);
    });

    // For Select2, use the 'change' event
    $('input[name="input2"]').on('change', function () {
        var textread = $(this).val();
        $('#result2').text(textread);
    });
});
```

**Reply by courtiol** [reply]
Thanks a lot, that now works as intended. I am getting one step closer to the use case.
